authors/apps/rating/tests/test_rating.py

```diff
@@ -0,0 +1,287 @@
+from django.urls import reverse
+from rest_framework import status
+from django.test import TestCase, Client
+import json
+from rest_framework.authtoken.models import Token
+from ..messages import error_msg, success_msg
+from authors.apps.articles.models import Article
+from authors.apps.authentication.models import User
+from authors.apps.authentication.backends import JWTokens
+
+
+class TestArticleRating(TestCase):
+    def setUp(self):
+        self.client = Client()
+
+        self.rating_body = {
+            'your_rating': 5
+        }
+
+        self.invalid_rating_body = {
+            'your_rating': "d"
+        }
+
+        self.create_user_1 = User.objects.create_user(
+            username="wachira",
+            email="ewachira254@gmail.com",
+            password="@Wachira254"
+        )
+        self.token_user_1 = JWTokens.create_token(
+            self, user=self.create_user_1)
+
+        self.create_user_2 = User.objects.create_user(
+            username="bolton",
+            email="bolton@gmail.com",
+            password="@bolton254"
+        )
+        self.token_user_2 = JWTokens.create_token(
+            self, user=self.create_user_2)
+
+        self.correct_article = Article.objects.create(
+            author_id=self.create_user_1.id,
+            image_path="...",
+            title="This is a blog",
+            description="This is a description",
+            body="This is a body",
+            slug="this-is-a-blog"
+        )
+
+    def test_post_rating(self):
+        url = reverse("rating:rate", kwargs={"slug": "this-is-a-blog"})
+        response = self.client.post(
+            url,
+            self.rating_body,
+            HTTP_AUTHORIZATION='Bearer ' + self.token_user_2,
+            content_type='application/json'
+        )
+        self.assertIn(response.data['message'], success_msg['rate_success'])
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertIn("your_rating", response.data['data'])
+
+        # Get Rating when logged in
+        url = reverse("rating:rate", kwargs={"slug": "this-is-a-blog"})
```

Comment:
Nice work on the tests 
```suggestion url = reverse("rating:rate", kwargs={"slug": "this-is-a-blog"})
        since this url is constant , you can define it as a global variable 
```

Reply:
This is done